src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -383,10 +384,7 @@ public interface StashAwareArtifactManager /* extends ArtifactManager */ {
         }
 
         private FilePath createTmpDir() throws IOException {
-            File dir = File.createTempFile("artifact", "copy");
-            if (!(dir.delete() && dir.mkdirs())) {
-                throw new IOException("Failed to create temporary directory " + dir.getPath());
-            }
+            File dir = Files.createTempDirectory("artifact" + "copy").toFile();
```

Comment:
i.e.
```suggestion
            File dir = Files.createTempDirectory("artifactcopy").toFile();
```

which was not the original intent. Something like this would be appropriate (generally like to switch to NIO).